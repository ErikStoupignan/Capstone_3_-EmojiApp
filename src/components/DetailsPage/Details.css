* {
  padding: 0;
  margin: 0;
}

:root {
  --square-dimensions: 170px;
  --backgroud-dark-template: #385894;
  --backgroud-light-template: #4369b2;
}

.MainContainer-Details {
  height: 99vh;
}

.Containerinfo-Details {
  height: 91vh;
}

/* Detailss Section */
.DetailsShow {
  height: 30%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.Details-Icon {
  width: 50%;
  font-size: 150px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding-bottom: 10px;
}

.DetailsRightContainer {
  width: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-right: 10px;
}

.Details-Name {
  font-size: 15px;
  color: white;
  font-weight: bold;
  text-align: right;
}

.Details-SubTitleGroup {
  padding: 0;
  margin: 0;
  font-size: 20px;
  color: white;
  font-weight: lighter;
  text-align: right;
}

.Ul-Details-Emojis {
  padding: 0;
  margin: 0;
  font-size: 20px;
  color: white;
  font-weight: lighter;
  text-align: left;
  height: 60%;
}

.Li-Details-Emojis {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 23%;
  padding: 0 5px 0 5px;
}

.Li-Details-Emojis:nth-child(even) {
  background: var(--backgroud-dark-template);
}

.Ul-placeholder {
  color: #8aa6d9;
}

.Details-SignalListOfEmojis {
  height: 3.5%;
  color: white;
  padding-left: 5%;
  background: var(--backgroud-dark-template);
  display: flex;
  align-items: center;
}

@media (min-width: 762px) {
  .MainContainer-Details {
    height: 100vh;
    width: 100vw;
  }

  .Containerinfo-Details {
    height: 90vh;
    display: flex;
    justify-content: space-between;
  }

  .DetailsShow {
    height: 100%;
    width: 46%;
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    justify-items: center;
  }

  .Details-Icon {
    grid-column: 1/2;
    grid-row: 1/2;
    width: 100%;
    height: 73%;
    padding-bottom: 110px;
    font-size: 30vw;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .DetailsRightContainer {
    grid-column: 1/2;
    grid-row: 1/2;
    width: 100%;
    height: 88%;
    display: flex;
    flex-direction: column;
    justify-content: end;
    padding: 0;
    margin: 0;
    position: relative;
  }

  .Details-Name {
    position: absolute;
    right: 15px;
    height: 100px;
    bottom: 15px;
    font-size: 30px;
    max-height: 150px;
    overflow-y: scroll;
    display: flex;
    align-items: center;
  }

  ::-webkit-scrollbar {
    display: none;
  }

  .Details-SubTitleGroup {
    position: absolute;
    top: 15px;
    left: 15px;
    font-size: 30px;
  }

  .Ul-Details-Emojis {
    height: 100%;
    width: 50%;
    padding: 0 15px 0 5px;
    font-size: 20px;
    color: white;
    font-weight: lighter;
    text-align: left;
  }

  .Li-Details-Emojis {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 19.8%;
    padding: 0 0 0 5px;
    font-size: 25px;
    width: 100%;
  }

  .Ul-property {
    width: 75%;
    height: 100%;
    display: flex;
    justify-content: left;
    align-items: center;
  }

  .Ul-placeholder {
    color: #8aa6d9;
    width: 18%;
    text-align: right;
    padding-right: 20px;
  }
}
