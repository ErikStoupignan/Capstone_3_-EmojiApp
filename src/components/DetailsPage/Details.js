import React from 'react';

const Details = () => (
  <div>This is Details Page!</div>
);

export default Details;
