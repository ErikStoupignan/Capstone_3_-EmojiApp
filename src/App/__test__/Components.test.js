import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import GetInformation from '../../components/HomePage/Home';
import Details from '../../components/DetailsPage/Details';
import store from '../../redux/configureStore';

describe('Testing the functions for Home Page', () => {
  it('Get information function', () => {
    const download = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <GetInformation />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(download).toMatchSnapshot();
  });
});

describe('Testing the functions for Details Page', () => {
  it('Details function', () => {
    const { getByText } = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Details />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(getByText('Ups... something was wrong')).toBeInTheDocument();
  });
});
