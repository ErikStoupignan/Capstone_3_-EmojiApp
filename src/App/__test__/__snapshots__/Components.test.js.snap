// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the functions for Home Page Get information function 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="Container-Home"
      >
        <nav
          class="Nav-bar"
        >
          <a
            class="NavBarDate-Arrow"
            href="/HomePage"
          >
            2022
          </a>
          <a
            class="NavBarTitle"
            href="https://github.com/ErikStoupignan"
            rel="noreferrer"
            target="_blank"
          >
            Emoji App
          </a>
          <div
            class="Navbar-IconsContainer"
          >
            <span
              class="Navbar-icon"
            >
              <a
                href="https://www.linkedin.com/in/erik-sanchez-beltran/"
                rel="noreferrer"
                target="_blank"
              />
            </span>
            <span
              class="Navbar-icon"
            >
              <a
                href="https://github.com/ErikStoupignan/Capstone_3_-EmojiApp"
                rel="noreferrer"
                target="_blank"
              />
            </span>
          </div>
        </nav>
        <div
          class="content-select"
        >
          <select
            class="menu-filter"
            name="select"
          >
            <option
              value="/all/"
            >
              üö© All Emojis üö©
            </option>
            <option
              value="/all/category_smileys_and_people"
            >
              üòé Smileys & People
            </option>
            <option
              value="/all/category_animals_and_nature"
            >
              üêò Animals & Nature
            </option>
            <option
              value="/all/category_food_and_drink"
            >
              üçï Food & Drink
            </option>
            <option
              value="/all/category_travel_and_places"
            >
              üõ´ Travel & Places
            </option>
            <option
              value="/all/category_activities"
            >
              üé≥ Activities
            </option>
            <option
              value="/all/category_objects"
            >
              üìö Objects
            </option>
            <option
              value="/all/category_symbols"
            >
              üö∏ Symbols
            </option>
          </select>
          <i>
            ‚§µ
          </i>
        </div>
        <div
          class="mediaQuery"
        >
          <div
            class="LoadingMessage"
          >
            <p>
              LOADING...
            </p>
            <p
              class="App-logo"
            >
              ü§∏
            </p>
          </div>
          <ul
            class="Ul-Emojis-Container"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="Container-Home"
    >
      <nav
        class="Nav-bar"
      >
        <a
          class="NavBarDate-Arrow"
          href="/HomePage"
        >
          2022
        </a>
        <a
          class="NavBarTitle"
          href="https://github.com/ErikStoupignan"
          rel="noreferrer"
          target="_blank"
        >
          Emoji App
        </a>
        <div
          class="Navbar-IconsContainer"
        >
          <span
            class="Navbar-icon"
          >
            <a
              href="https://www.linkedin.com/in/erik-sanchez-beltran/"
              rel="noreferrer"
              target="_blank"
            />
          </span>
          <span
            class="Navbar-icon"
          >
            <a
              href="https://github.com/ErikStoupignan/Capstone_3_-EmojiApp"
              rel="noreferrer"
              target="_blank"
            />
          </span>
        </div>
      </nav>
      <div
        class="content-select"
      >
        <select
          class="menu-filter"
          name="select"
        >
          <option
            value="/all/"
          >
            üö© All Emojis üö©
          </option>
          <option
            value="/all/category_smileys_and_people"
          >
            üòé Smileys & People
          </option>
          <option
            value="/all/category_animals_and_nature"
          >
            üêò Animals & Nature
          </option>
          <option
            value="/all/category_food_and_drink"
          >
            üçï Food & Drink
          </option>
          <option
            value="/all/category_travel_and_places"
          >
            üõ´ Travel & Places
          </option>
          <option
            value="/all/category_activities"
          >
            üé≥ Activities
          </option>
          <option
            value="/all/category_objects"
          >
            üìö Objects
          </option>
          <option
            value="/all/category_symbols"
          >
            üö∏ Symbols
          </option>
        </select>
        <i>
          ‚§µ
        </i>
      </div>
      <div
        class="mediaQuery"
      >
        <div
          class="LoadingMessage"
        >
          <p>
            LOADING...
          </p>
          <p
            class="App-logo"
          >
            ü§∏
          </p>
        </div>
        <ul
          class="Ul-Emojis-Container"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
